### PR TITLE
Improve and document integration tests

### LINKED ISSUE #914: Integration and compatibility tests improvements
* #983
* ~support documents without "value" field ("unset")~ moved to #1023
* ~use read-only user where possible~ moved to #1025
* #862
* #984
* add more composite data sets
* document compatibility tests

### PR DESCRIPTION
**Pull request by AlekSi**
Closes #914.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [x] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by AlekSi** [comment]
hm…

**Comment by rumyantseva** [comment]
@AlekSi Tigris doesn't have read only, right? So, this will only apply for Postgres?

File: integration/setup/common.go [hunk 1]
```diff
@@ -63,14 +63,20 @@ func SkipForTigris(tb testing.TB) {
 
 // setupListener starts in-process FerretDB server that runs until ctx is done,
 // and returns listening port number.
-func setupListener(tb testing.TB, ctx context.Context, logger *zap.Logger) int {
+func setupListener(tb testing.TB, ctx context.Context, logger *zap.Logger, readOnly bool) int {
 	tb.Helper()
 
+	if *handlerF == "tigris" && readOnly {
+		tb.Logf("Read-only user is not implemented for %q handler yet.", *handlerF)
```
Comment (AlekSi) [inline integration/setup/common.go:70]: @rumyantseva

Comment (rumyantseva) [reply integration/setup/common.go:70]: Ah!

**Review by rumyantseva — DISMISSED** [review]
LGTM

File: integration/update_field_test.go [hunk 2]
```diff
@@ -1098,11 +1098,12 @@ func TestUpdateFieldSetOnInsert(t *testing.T) {
 			}
 
 			require.NoError(t, err)
+
+			expectedStat := tc.expectedStat
 			if tc.upserted {
-				tc.expectedStat.UpsertedID = tc.id
```
Comment (AlekSi) [inline integration/update_field_test.go:1102]: There was a data race

File: cmd/envtool/main.go [hunk 3]
```diff
@@ -199,25 +199,27 @@ func setupPostgres(ctx context.Context, logger *zap.SugaredLogger) error {
...
+
 		`ANALYZE`, // to make tests more stable
 	} {
 		if _, err = pgPool.Exec(ctx, q); err != nil {
 			return err
 		}
 	}
 
+	logger.Info("Creating databases...")
+	for _, db := range []string{"admin", "test"} {
```
Comment (noisersup) [inline cmd/envtool/main.go:217]:
```suggestion

	for _, db := range []string{"admin", "test"} {
```
linter fails

**Review by noisersup — CHANGES_REQUESTED** [review]
The linter fails, but overall LGTM

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by noisersup — APPROVED** [review]

**Pull request merged**